backend/src/core/models/game.spec.js

```diff
@@ -1,583 +1,779 @@
+import { getParsedHtmlPage } from "../../../assets/html.details.pages.mock.js";
+import { crusaderKingsDetailsPage } from "../../../assets/steam-details-pages/crusader.kings.multiple.developers.html.details.page.js";
+import { feartressGameHtmlDetailsPage } from "../../../assets/steam-details-pages/feartress.game.html.details.page.js";
+import { mortalDarknessGameHtmlDetailsPage } from "../../../assets/steam-details-pages/mortal.darkness.game.html.details.page.js";
+import { riskOfRainHtmlDetailsPageMissingInfo } from "../../../assets/steam-details-pages/risk.of.rain.missing.additional.info.page.js";
+import { counterStrikeHtmlDetailsSteamDb } from "../../../assets/steamdb-details-pages/counter.strike.html.details.page.js";
+import { eldenRingHtmlDetailsPageDb } from "../../../assets/steamdb-details-pages/elden.ring.html.details.page.js";
+import { karmazooHtmlDetailsPageSteamDb } from "../../../assets/steamdb-details-pages/karmazoo.html.details.page.js";
+import { riskOfRainHtmlDetailsSteamDb } from "../../../assets/steamdb-details-pages/risk.of.rain.html.details.page.js";
 import { Game } from "./game.js";
-import { getOneSteamAppInstantiatedGame, getXGamesWithoutDetails } from "./game.mocks.js";
+import { getOneGameWithPlayerHistory, getXGamesWithoutDetails } from "./game.mocks.js";
 import { PlayerHistory } from "./player.history.js";
 import { getXSampleSteamApps } from "./steam.app.mocks.js";
 
-describe("game.js", function () {
-  describe("Game", function () {
-    describe(".fromSteamApp", function () {
-      describe("is called with no arguments, ", function () {
-        it("an Error is thrown", function () {
-          expect(Game.fromSteamApp).toThrowError();
-        });
-      });
+describe("Game", function () {
+  describe(".copy", function () {
+    beforeAll(function () {
+      this.game = getOneGameWithPlayerHistory();
+      this.result = this.game.copy();
 
-      describe("is called with incomplete arguments, ", function () {
-        beforeAll(function () {
-          this.testObject = {
-            id: 123,
-            name: "test game",
-          };
-        });
+      this.game.id = 99;
+    });
 
-        it("an Error is thrown", function () {
-          expect(Game.fromSteamApp.bind(this.testObject)).toThrowError();
-        });
+    it("the result is a copy of game", function () {
+      expect(this.result).toBeInstanceOf(Game);
+      expect(this.result.playerHistory[0]).toBeInstanceOf(PlayerHistory);
+    });
+
+    it("the result has the correct values", function () {
```

Comment:
`the copy has identical values to the original

Reply:
Just to note here, it doesn't have identical values to the original, as I change one of the values of the original to see if the value in the copy would change as well, meaning I'm testing for whether or not its a deep copy (the changing of a value won't change the copy also).